test/README.md

```diff
@@ -0,0 +1,124 @@
+# Testing strategy
+
+This document describes the desired testing strategy for the project, which should
+slowly be implemented over time.
+
+This is a live document and should be updated as the project evolves, to cover
+details like `api` testing or `benchmarking`.
+
+## Introduction
+
+This project is a complex system with many moving parts. It is important to have a
+coherent testing strategy to ensure that the system works as expected,
+doesn't break over time, and very importantly, make sure that developers know
+where and how to test things.
+
+As testing tools we use two frameworks:
+
+- Ginkgo and Gomega, a BDD-style testing framework for Go, used for more complex
+  testing where BeforeEach/AfterEach is beneficial. It has de benefit of allowing
+  to write tests in a more human-readable way.
+
+- Go's built-in testing framework with testify, used for simpler unit tests.
+
+## Internal test libraries
+
+We maintain our internal test libraries in `/test/pkg/`
+
+# Types of tests
+
+## Unit testing
+
+Unit tests are used to test individual functions or methods in isolation. They
+should be fast, and should not depend on external services or databases.
+
+We keep the test files for unit tests in the same directory as the code we are
+testing.
+
+i.e.
+
+code in `pkg/log/log.go` should have unit tests in `pkg/log/log_test.go`
+
+We generally use testify, and sometimes Ginkgo/gomega, depending on the nature
+of the test, the choice here is flexible.
+
+Unit tests can be run locally with:
+    
+```bash
+make unit-test # or run-unit-test if you have a DB server ready
+```
+
+### Mocking
+
+Sometimes we need to mock interfaces to make unit testing possible in
+isolation. For that we use the `mockgen` tool from `go.uber.org/mock`
+
+If you want to generate mocks for a package, you can add the reference to
+`/hack/mock.list.txt` and run `make generate` to generate the mocks.
+
+Find more information about using mockgen [here](https://pkg.go.dev/go.uber.org/mock#readme-building-mocks)
+
+
+## Integration testing
```

Comment:
Lets elaborate (in few sentences) what is the expected different between integration and e2e? what components will be running in integration? will it also be deployed on kind? or mostly mocked as running binaries? Other differences?